modules/metastorage/README.md

```diff
@@ -0,0 +1,40 @@
+# Metastorage
+
+The module for store and access to metadata.
+
+The storage is distributed to prevent losing data and based of RAFT. In typical case, the group consists of only several nodes of the
+cluster (amount of nodes have to be odd 3 or 5). The rest of nodes listens metadata update but does not participant in voting (RAFT
+listeners).
+
+## Threading model
+
+There is only one dedicate thread to notify watches. This is a thread with prefix metastorage-watch-executor.
+The thread is created from executor in KeyValueStorage:
+
+```java
+private final ExecutorService watchExecutor = Executors.newSingleThreadExecutor(new NamedThreadFactory("metastorage-watch-executor",LOG));
+```
+
+Also, the storage internally contains a two threads executor to create a snapshot:
+
+```java
+private final ExecutorService snapshotExecutor = Executors.newFixedThreadPool(2,new NamedThreadFactory("metastorage-snapshot-executor",LOG));
+```
+
+Both of the executors have no a node name in their names. The names are required to identify threads belonged to different Ignite nodes on
+the shared java machine (TODO: IGNITE-18504 Add nodes name in prefix to Metastorage executors).
+
+### Interface methods
```

Comment:
Initially, there were asyncs on cursors, we do expect this to be reworked with subscribers, correct? If true this subscribers will probably add some (one?) new threads. Could you please create a follow-up ticket in order to update ms thread-model after Alexander will finish his ms-cursors refactoring?

Reply:
We even have no an JIRA ticket to redesign cursor API to reactive model for Metastorage. But I notified @sashapolo about the modification can be required after the API will have changed.